src/core/evaluator.js

```diff
@@ -2109,27 +2129,24 @@ class PartialEvaluator {
       // Replaces all whitespaces with standard spaces (0x20), to avoid
       // alignment issues between the textLayer and the canvas if the text
       // contains e.g. tabs (fixes issue6612.pdf).
-      var i = 0,
-        ii = str.length,
-        code;
-      while (i < ii && (code = str.charCodeAt(i)) >= 0x20 && code <= 0x7f) {
-        i++;
-      }
-      return i < ii ? str.replace(WhitespaceRegexp, " ") : str;
+      return str.replace(BeforeSpaceRegexp, " ");
```

Comment:
Wait, is this actually equivalent to the old code though?

The old `WhitespaceRegexp = /\s/g` includes *a lot* more things than the new `BeforeSpaceRegexp` does, according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Cheatsheet:
> Matches a single white space character, including space, tab, form feed, line feed, and other Unicode spaces. Equivalent to [ \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]. For example, /\s\w*/ matches " bar" in "foo bar".

I'd suggest reverting this change, to avoid unforeseen regressions!